### ISSUE TITLE
Support exclusion strategy for classes to be skipped from serialization

### ISSUE THREAD

**Issue by gpor0** [javers/javers#940]
Javers supports registerIgnoredClass API to ignore specific class. It would be very useful in some cases to support ignoreStrategy / exclusionStrategy where implementor can define custom rule like ignoring by package or similar.

Gson already supports exclusion strategy but as far as i am aware of Gson configuration in Javers cannot be accessed at the moment.

**Comment by bartoszwalacik** [comment]
Makes sense, we will accept your PR if you like to contribute.

API sketch :

```
javersBuilder.registerIgnoredClassesStrategy(IgnoredClassesStrategy strategy)
```

```
interface IgnoredClassesStrategy {

    boolen isIgnored(Class<?> domainClass);

}
```

**Comment by bartoszwalacik** [comment]
```java
Javers javers = JaversBuilder.javers()
        .registerIgnoredClassesStrategy(c -> c.getName().startsWith("com.ignore.me"))
        .build(); 
```

**Comment by bartoszwalacik** [comment]
released in 5.11.1
